libs/balances-and-allowances/src/hooks/usePersistBalancesFromBff.ts

```diff
@@ -0,0 +1,116 @@
+import { useSetAtom } from 'jotai'
+import { useEffect } from 'react'
+
+import { BFF_BASE_URL } from '@cowprotocol/common-const'
+import { useDebounce } from '@cowprotocol/common-hooks'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { useWalletInfo } from '@cowprotocol/wallet'
+import { BigNumber } from '@ethersproject/bignumber'
+
+import ms from 'ms.macro'
+import useSWR, { SWRConfiguration } from 'swr'
+
+import { balancesAtom, BalancesState, balancesUpdateAtom } from '../state/balancesAtom'
+import { useSetIsBffFailed } from '../state/isBffFailedAtom'
+
+type BalanceResponse = {
+  balances: Record<string, string> | null
+}
+
+export interface PersistBalancesFromBffParams {
+  account?: string
+  chainId: SupportedChainId
+  balancesSwrConfig: SWRConfiguration
+  pendingOrdersCount?: number
+  tokenAddresses: string[]
+}
+
+const DEBOUNCE_FOR_PENDING_ORDERS_MS = ms`1s`
+
+export function usePersistBalancesFromBff(params: PersistBalancesFromBffParams): void {
+  const { account, chainId, balancesSwrConfig, pendingOrdersCount, tokenAddresses } = params
+
+  const { chainId: activeChainId, account: connectedAccount } = useWalletInfo()
+  const targetAccount = account ?? connectedAccount
+  const targetChainId = chainId ?? activeChainId
+
+  const setIsBffFailed = useSetIsBffFailed()
+  const debouncedPendingOrdersCount = useDebounce(pendingOrdersCount, DEBOUNCE_FOR_PENDING_ORDERS_MS)
+
+  const {
+    isLoading: isBalancesLoading,
+    data,
+    error,
+  } = useSWR(
+    // debouncedPendingOrdersCount is added to the key to refetch balances when it changes (new order created or order filled)
+    targetAccount ? [targetAccount, targetChainId, debouncedPendingOrdersCount, 'bff-balances'] : null,
+    ([walletAddress, chainId, _]) => getBffBalances(walletAddress, chainId),
```

Comment:
Nitpick: don't need to de-structure variables that not used, no?


```suggestion
    ([walletAddress, chainId]) => getBffBalances(walletAddress, chainId),
```